src/scalar_planner_estimates.rs

```diff
@@ -0,0 +1,220 @@
+    BUTTERFLY_SLOPE_16 * repeats as f32 + BUTTERFLY_CONST_16
+}
+
+const BUTTERFLY_SLOPE_17: f32 = 21.93296;
+const BUTTERFLY_CONST_17: f32 = -5.98342;
+
+pub fn estimate_butterfly_cost_17(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_17 * repeats as f32 + BUTTERFLY_CONST_17
+}
+
+const BUTTERFLY_SLOPE_19: f32 = 16.20785;
+const BUTTERFLY_CONST_19: f32 = -0.16664;
+
+pub fn estimate_butterfly_cost_19(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_19 * repeats as f32 + BUTTERFLY_CONST_19
+}
+
+const BUTTERFLY_SLOPE_23: f32 = 41.35303;
+const BUTTERFLY_CONST_23: f32 = -1.76352;
+
+pub fn estimate_butterfly_cost_23(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_23 * repeats as f32 + BUTTERFLY_CONST_23
+}
+
+const BUTTERFLY_SLOPE_29: f32 = 84.98456;
+const BUTTERFLY_CONST_29: f32 = -1.58332;
+
+pub fn estimate_butterfly_cost_29(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_29 * repeats as f32 + BUTTERFLY_CONST_29
+}
+
+const BUTTERFLY_SLOPE_31: f32 = 100.00000;
+const BUTTERFLY_CONST_31: f32 = -4.88995;
+
+pub fn estimate_butterfly_cost_31(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_31 * repeats as f32 + BUTTERFLY_CONST_31
+}
+
+const BUTTERFLY_SLOPE_32: f32 = 36.36041;
+const BUTTERFLY_CONST_32: f32 = -0.08669;
+
+pub fn estimate_butterfly_cost_32(repeats: usize) -> f32 {
+    BUTTERFLY_SLOPE_32 * repeats as f32 + BUTTERFLY_CONST_32
+}
+
+// --- End code generated by tools/estimates/fit_butterflies.py ---
+
+// --- Begin code generated by tools/estimates/fit_radix4.py ---
+
+const RADIX4_CONST: f32 = 0.90900;
+const RADIX4_SLOPE: f32 = 0.58761;
+const RADIX4_EXP: f32 = 1.15980;
+
+pub fn estimate_radix4_cost(len: usize, repeats: usize) -> f32 {
+    (RADIX4_CONST + RADIX4_SLOPE * (len as f32).powf(RADIX4_EXP)) * repeats as f32
+}
+
+// --- End code generated by tools/estimates/fit_radix4.py ---
+
+// --- Begin code generated by tools/estimates/fit_raders.py ---
+
+const RADERS_CONST: f32 = -1.22742;
+const RADERS_SLOPE: f32 = 2.30372;
+
+pub fn estimate_raders_cost(inner_fft: &Arc<Recipe>, repeats: usize) -> f32 {
+    (RADERS_CONST + RADERS_SLOPE * (inner_fft.len() + 1) as f32 + 2.0 * inner_fft.cost(1))
+        * repeats as f32
+}
+
+// --- End code generated by tools/estimates/fit_raders.py ---
+
+// --- Begin code generated by tools/estimates/fit_bluesteins.py ---
+
+const BLUESTEINS_CONST: f32 = 16.77562;
+const BLUESTEINS_LEN_SLOPE: f32 = 0.30714;
+const BLUESTEINS_INNER_LEN_SLOPE: f32 = 0.34986;
+
+pub fn estimate_bluesteins_cost(len: usize, inner_fft: &Arc<Recipe>, repeats: usize) -> f32 {
+    (BLUESTEINS_CONST
+        + BLUESTEINS_INNER_LEN_SLOPE * inner_fft.len() as f32
+        + BLUESTEINS_LEN_SLOPE * len as f32
+        + 2.0 * inner_fft.cost(1))
+        * repeats as f32
+}
+
+// --- End code generated by tools/estimates/fit_bluesteins.py ---
+
+// --- Begin code generated by tools/estimates/fit_mixedradixes.py ---
+
+const MIXEDRADIX_SLOPE: f32 = 0.09496;
+const MIXEDRADIX_EXP: f32 = 1.18301;
```

Comment:
Something that occurs to me - these curves suggest that the performance of these algorithms is O(N^X), with X=1.18. But we know that that's not true, FFT algorithms are N(nlogn). I haven't looked into how you did curve fitting yet, so maybe this is harder than it sounds, but is it possible to do a curve fitting with A * N * log(B * N) instead?

Reply:
I guess the most complete form would be A + N * (B + C * Log(D + E * N)), but 5 variables might be too many to fit.

Reply:
I was also surprised by this result. I first tried with NlogN but it just didn't fit very well. My guess is that while it's NlogN in number of calculations, there is also the cache and memory systems that make things slow down at long lengths.

Reply:
![rx4fit](https://user-images.githubusercontent.com/6504678/108051890-b7984580-704b-11eb-9c7e-ab8e539b7859.png)
This is the fitting result that made the coeffiecients in the PR.